manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,639 @@
+            return false;
+        }
+                // IMEI should already be checked
+        return  Objects.equals(topic.getTokens().get(2), TELTONIKA_DEVICE_TOKEN) &&
+                IMEIValidator.isValidIMEI(imeiValue) &&
+                (
+                        Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_RECEIVE_TOPIC) ||
+                        Objects.equals(topic.getTokens().get(4), TELTONIKA_DEVICE_SEND_TOPIC)
+                );
+    }
+
+    /**
+     * Overrides MQTTHandler.checkCanPublish for this specific Handler,
+     * until secure Authentication and Auto-provisioning
+     * of Teltonika Devices is created.
+     * To be removed after implementation is complete.
+     */
+    @Override
+    public boolean checkCanPublish(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        return true;
+    }
+
+    @Override
+    public boolean canPublish(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        getLogger().info("Teltonika device will publish to Topic "+topic.toString()+" to transmit payload");
+        return true;
+    }
+
+    public void onSubscribe(RemotingConnection connection, Topic topic) {
+        getLogger().info("CONNECT: Device "+topic.tokens.get(1)+" connected to topic "+topic+".");
+        connectionSubscriberInfoMap.put(topic.tokens.get(3), new TeltonikaDevice(topic));
+    }
+
+    @Override
+    public void onUnsubscribe(RemotingConnection connection, Topic topic) {
+        getLogger().info("DISCONNECT: Device "+topic.tokens.get(1)+" disconnected from topic "+topic+".");
+        connectionSubscriberInfoMap.remove(topic.tokens.get(3));
+    }
+
+    /**
+     * Get the set of topics this handler wants to subscribe to for incoming publish messages; messages that match
+     * these topics will be passed to {@link #onPublish}.
+     * The listener topics are defined as <code>{realmID}/{userID}/{@value TOKEN_TELTONIKA_DEVICE}/{IMEI}</code>
+     * //DONE: Be explicit about sending data to {IMEI}/data, and sending commands to {IMEI}/commands.
+     * //TODO: Understand the data flow for which topics should be subscribed/not subscribed by the handler. I think that the command topic should not be there since we're not looking to read the command we just sent.
+     */
+    @Override
+    public Set<String> getPublishListenerTopics() {
+        return Set.of(
+                TOKEN_SINGLE_LEVEL_WILDCARD + "/" + TOKEN_SINGLE_LEVEL_WILDCARD + "/" +
+                        TOKEN_TELTONIKA_DEVICE + "/" + TOKEN_SINGLE_LEVEL_WILDCARD + "/" + TELTONIKA_DEVICE_RECEIVE_TOPIC,
+                TOKEN_SINGLE_LEVEL_WILDCARD + "/" + TOKEN_SINGLE_LEVEL_WILDCARD + "/" +
+                        TOKEN_TELTONIKA_DEVICE + "/" + TOKEN_SINGLE_LEVEL_WILDCARD + "/" + TELTONIKA_DEVICE_SEND_TOPIC
+        );
+    }
+
+    @Override
+    public void onPublish(RemotingConnection connection, Topic topic, ByteBuf body) {
```

Comment:
Please add some comments / context to this method.
I'm not fully known with the use case, but for me it's weird to create assets during `onPublish()` ?
Especially since no actual payload is published?